src/daft-connect/src/lib.rs

```diff
@@ -0,0 +1,216 @@
+#![feature(iterator_try_collect)]
+#![feature(let_chains)]
+#![feature(try_trait_v2)]
+#![feature(coroutines)]
+#![feature(iter_from_coroutine)]
+#![feature(stmt_expr_attributes)]
+#![feature(try_trait_v2_residual)]
+#![warn(unused)]
+
+use dashmap::DashMap;
+use eyre::Context;
+#[cfg(feature = "python")]
+use pyo3::types::PyModuleMethods;
+use spark_connect::{
+    spark_connect_service_server::{SparkConnectService, SparkConnectServiceServer},
+    AddArtifactsRequest, AddArtifactsResponse, AnalyzePlanRequest, AnalyzePlanResponse,
+    ArtifactStatusesRequest, ArtifactStatusesResponse, ConfigRequest, ConfigResponse,
+    ExecutePlanRequest, ExecutePlanResponse, FetchErrorDetailsRequest, FetchErrorDetailsResponse,
+    InterruptRequest, InterruptResponse, ReattachExecuteRequest, ReleaseExecuteRequest,
+    ReleaseExecuteResponse, ReleaseSessionRequest, ReleaseSessionResponse,
+};
+use tonic::{transport::Server, Request, Response, Status};
+use tracing::{info, warn};
+use uuid::Uuid;
+
+use crate::session::Session;
+
+mod config;
+mod err;
+mod session;
+pub mod util;
+
+pub fn start(addr: &str) -> eyre::Result<()> {
```

Comment:
I think we should have a stateful `Connection` object that is returned to python instead of `Result<()>` 

I was trying this out locally in a notebook, but quickly ran into issues with the port being used as there's no way to gracefully close the connection.

Reply:
added